src/requireHooks.ts

```diff
@@ -1,8 +1,11 @@
 import logger from '@pnpm/logger'
+import path = require('path')
+import pathAbsolute = require('path-absolute')
 import requirePnpmfile from './requirePnpmfile'
 
-export default function requireHooks (prefix: string) {
-  const pnpmFile = requirePnpmfile(prefix)
+export default function requireHooks (prefix: string, customPnpmfileLocation: string | undefined) {
+  const pnpmFile = requirePnpmfile(path.join(prefix, 'pnpmfile.js'))
+    || customPnpmfileLocation && requirePnpmfile(pathAbsolute(customPnpmfileLocation, prefix))
```

Comment:
I don't know the code, but shouldn't a custom location take precedence over the standard location?

Reply:
Hard to answer. I think it would be a breaking change. Someone created a pnpmfile in the root of their package and expected it to be picked up. Someone else sets the pnpmfile globally and the `pnpmfile.js` in the root of the projects gets ignored now?

I think in pnpm v2 we should only read the pnpmfile if there is this config. Otherwise, it is hard too complex.

Reply:
I really think merging the result of the different files in a well defined order would be best.

Reply:
ok, I'll do it. They will be composed

Reply:
I would not want to add additional filesystem operations for searching the global pnpmfile. However, we can have an additional config, `global-pnpmfile`. In that case, we can read both the global pnpmfile and the local one and compose them. Sounds good?

Reply:
Seems a fair compromise.